packages/autocomplete-core/src/__tests__/getAutocompleteSetters.test.ts

```diff
@@ -51,19 +54,56 @@ describe('createAutocomplete', () => {
     });
   });
 
-  test('setSuggestions', () => {
-    const onStateChange = jest.fn();
-    const { setSuggestions } = createAutocomplete({
-      getSources: () => [],
-      onStateChange,
+  describe('setSuggestions', () => {
+    test('default', () => {
+      const onStateChange = jest.fn();
+      const { setSuggestions } = createAutocomplete({
+        getSources: () => [],
+        onStateChange,
+      });
+
+      setSuggestions([createSuggestion([{ label: 'hi' }])]);
+
+      expect(onStateChange).toHaveBeenCalledTimes(1);
+      expect(onStateChange).toHaveBeenCalledWith({
+        state: expect.objectContaining({
+          suggestions: [
+            {
+              items: [
+                {
+                  label: 'hi',
+                  __autocomplete_id: 0,
+                },
+              ],
+              source: expect.any(Object),
+            },
+          ],
+        }),
+      });
     });
-    const suggestions = [createSuggestion()];
 
-    setSuggestions(suggestions);
-
-    expect(onStateChange).toHaveBeenCalledTimes(1);
-    expect(onStateChange).toHaveBeenCalledWith({
-      state: expect.objectContaining({ suggestions }),
+    test('flattens suggestions', () => {
+      const onStateChange = jest.fn();
+      const { setSuggestions } = createAutocomplete({
+        openOnFocus: true,
+        getSources: () => [],
+        onStateChange,
+      });
+
+      // @ts-expect-error AutocompleteSuggestion has an array of items
+      // while it also accepts a nested array.
+      // There's only one type for both input and output however, 
+      setSuggestions([createSuggestion([[{ label: 'hi' }]])]);
```

Comment:
do you know how to fix this @francoischalifour? I don't see how I can setSuggestions allow a nested array in a way where the internal type has the flattened version

Reply:
It should be here: https://github.com/algolia/autocomplete.js/blob/367389ed427e603450ddb74d1f40e151711592f6/packages/autocomplete-core/src/types/setters.ts#L10

Reply:
do you mean to make a new type for AutocompleteState that's the input? I'm not sure I get your suggestion

Reply:
I mean that it should be updated to:

```ts
setSuggestions: StateUpdater<AutocompleteState<TItem>['suggestions'] | Array<AutocompleteState<TItem>['suggestions']>>; 
```

Unless I misunderstood your question?

Reply:
that's not the issue, since we need to accept an array of items, or a nested array of items, this is an array of suggestions

Reply:
Can't you use the return type of `getSuggestions` in this `setSuggestions` type?